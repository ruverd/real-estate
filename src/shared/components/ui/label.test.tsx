import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { describe, expect, it, vi } from "vitest";

import { Label } from "./label";

describe("Label", () => {
  it("should renders correctly", () => {
    render(<Label>Test Label</Label>);

    expect(screen.getByText("Test Label")).toBeInTheDocument();
  });

  it("should applies custom className when provided", () => {
    render(<Label className="custom-class">Test Label</Label>);

    expect(screen.getByText("Test Label")).toHaveClass("custom-class");
  });

  it("should forwards the htmlFor attribute correctly", () => {
    render(<Label htmlFor="test-input">Test Label</Label>);

    expect(screen.getByText("Test Label")).toHaveAttribute("for", "test-input");
  });

  it("should handles click events", async () => {
    const handleClick = vi.fn();

    render(<Label onClick={handleClick}>Clickable Label</Label>);
    
    const label = screen.getByText("Clickable Label");
    await userEvent.click(label);
    
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it("should renders with additional attributes", () => {
    render(<Label data-testid="test-label">Test Label</Label>);

    expect(screen.getByTestId("test-label")).toBeInTheDocument();
  });

  it("should matches snapshot", () => {
    const { container } = render(<Label>Test Label</Label>);

    expect(container).toMatchSnapshot();
  });
});
