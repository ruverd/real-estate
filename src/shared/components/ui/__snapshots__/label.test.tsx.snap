// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Label > should matches snapshot 1`] = `
<div>
  <label
    class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50"
    data-slot="label"
  >
    Test Label
  </label>
</div>
`;
